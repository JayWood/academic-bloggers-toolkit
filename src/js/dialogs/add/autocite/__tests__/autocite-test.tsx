import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';
import AutoCite from '..';

const getter = jest.fn();

const setup = ({ kind = 'webpage' } = {}, pattern?: any) => {
    const component = mount(
        <AutoCite kind={kind as any} placeholder="Test" pattern={pattern} getter={getter} />,
    );
    return {
        component,
    };
};

describe('<AutoCite />', () => {
    beforeEach(() => jest.resetAllMocks());
    test('should match snapshots', () => {
        let { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup({ kind: 'book' }, '[0-9]+'));
        expect(toJSON(component)).toMatchSnapshot();
    });
    test('should handle field change', () => {
        const { component } = setup();
        const query = component.find('#citequery');
        query.simulate('change', { currentTarget: { value: 'hello' } });
    });
    test('should call handleQuery', () => {
        const { component } = setup();
        const instance = (component as any).instance();
        const button = component.find('button');
        instance.input = {
            validity: {
                valid: true,
            },
        };
        instance.query.set('testing');
        button.simulate('click');

        expect(instance.query.get()).toBe('');
        expect(getter).toHaveBeenCalledTimes(1);

        instance.input.validity = {
            valid: false,
        };
        instance.query.set('foo');
        button.simulate('click');

        expect(instance.query.get()).toBe('foo');
        expect(getter).toHaveBeenCalledTimes(1);
    });
    test('should call handleKeyDown', () => {
        const { component } = setup();
        const input = component.find('#citequery');
        const stopPropagation = jest.fn();
        const preventDefault = jest.fn();

        input.simulate('keyDown', { key: 'Enter', stopPropagation, preventDefault });
        input.simulate('keyDown', { key: 'Return', stopPropagation, preventDefault });
        input.simulate('keyDown', { key: 'a', stopPropagation, preventDefault });

        expect(stopPropagation).toHaveBeenCalledTimes(2);
        expect(preventDefault).toHaveBeenCalledTimes(2);
    });
});
