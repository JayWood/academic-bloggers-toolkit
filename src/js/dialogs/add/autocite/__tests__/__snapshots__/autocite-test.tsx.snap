// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoCite /> should match snapshots 1`] = `
<AutoCite
  getter={[MockFunction]}
  kind="webpage"
  placeholder="Test"
>
  <div
    className="autocite"
  >
    <label
      htmlFor="citequery"
    >
      Autocite
    </label>
    <input
      className="input"
      id="citequery"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Test"
      type="url"
      value=""
    />
    <Button
      disabled={true}
      flat={true}
      label="Search"
      onClick={[Function]}
      type="button"
    >
      <div>
        <button
          className="btn btnFlat"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </Button>
  </div>
</AutoCite>
`;

exports[`<AutoCite /> should match snapshots 2`] = `
<AutoCite
  getter={[MockFunction]}
  kind="book"
  pattern="[0-9]+"
  placeholder="Test"
>
  <div
    className="autocite"
  >
    <label
      htmlFor="citequery"
    >
      Autocite
    </label>
    <input
      className="input"
      id="citequery"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern="[0-9]+"
      placeholder="Test"
      type="text"
      value=""
    />
    <Button
      disabled={true}
      flat={true}
      label="Search"
      onClick={[Function]}
      type="button"
    >
      <div>
        <button
          className="btn btnFlat"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </Button>
  </div>
</AutoCite>
`;
